idioms/default.md

```diff
@@ -0,0 +1,50 @@
+# The `Default` Trait
+
+## Description
+
+Many types in Rust have a [Constructor]. However, this is *specific* to the
+type; Rust cannot abstract over "everything that has a `new()` method". To
+allow this, the [`Default`] trait was conceived, which can be used with
+containers and other generic types (e.g. see [`Option::unwrap_or_default()`]).
+Notably, some containers already implement it where applicable.
+
+Not only do one-element containers like `Cow`, `Box` or `Arc` implement
+`Default` for contained `Default` types, one can automatically
+`#[derive(Default)]` for structs whose fields all implement it, so the more
+types implement `Default`, the more useful it becomes.
+
+## Example
+
+```rust
+// note that we can simply auto-derive Default here.
+#[derive(Default)]
+struct Interesting {
```

Comment:
`Default` is often used for configuration structs -- I'd prefer if we use a more configuration-like theme for the example.